typescript/phoenix-sdk/tests/swap.ts

```diff
@@ -137,14 +137,71 @@ export async function swap() {
 
   let txResult = await Phoenix.getEventsFromTransaction(connection, txId);
   let counter = 1;
-  while (txResult.instructions.length == 0) {
+  while (!txResult.txReceived) {
     txResult = await Phoenix.getEventsFromTransaction(connection, txId);
     counter += 1;
     if (counter == 10) {
       throw Error("Failed to fetch transaction");
     }
   }
 
+  if (txResult.failed) {
+    console.log("Swap transaction failed");
+    return;
+  }
+
+  // Really dirty test for expired TIF
+  const expiredOrderPacket = Phoenix.getMarketSwapOrderPacketWithTimeInForce({
+    marketData,
+    side,
+    inAmount,
+    lastValidUnixTimestampInSeconds: 10, // This time will be treated as expired
+    lastValidSlot: null,
+    slippage,
+  });
+
+  const expiredSwapIx = Phoenix.createSwapInstruction(orderAccounts, {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore TODO why is __kind incompatible?
+    orderPacket: {
+      __kind: "ImmediateOrCancel",
+      ...expiredOrderPacket,
+    },
+  });
+
+  const expiredSwapTx = new Transaction().add(expiredSwapIx);
+  const expiredTxId = await connection.sendTransaction(
+    expiredSwapTx,
+    [trader],
+    {
+      skipPreflight: true,
+    }
+  );
+  await connection.confirmTransaction(expiredTxId, "confirmed");
+  console.log("Transaction ID:", expiredTxId);
+
+  let expiredTxResult = await Phoenix.getEventsFromTransaction(
+    connection,
+    expiredTxId
+  );
+  let expiredCounter = 1;
+  while (!expiredTxResult.txReceived) {
+    console.log(expiredTxResult);
+    expiredTxResult = await Phoenix.getEventsFromTransaction(
+      connection,
+      expiredTxId
+    );
+    expiredCounter += 1;
+    if (expiredCounter == 10) {
+      throw Error("Failed to fetch transaction");
+    }
+  }
+
+  if (expiredTxResult.failed) {
+    console.log("Expired transaction not supposed to fail");
```

Comment:
why not `throw Error` for consistency? or `console.error`

Reply:
test is non deterministic because the market in the background is changing, IMO it's okay to not crash if the tx returns a failure